diff --git a/taglist.vim b/taglist.vim
index 59901f6..3e0593b 100644
--- a/taglist.vim
+++ b/taglist.vim
@@ -969,7 +969,7 @@ function! s:Tlist_Detect_Filetype(fname)
     exe 'doautocmd filetypedetect BufRead ' . a:fname
 
     " Save the detected filetype
-    let ftype = &filetype
+    let ftype = matchstr(&filetype, '[^.]\+')
 
     " Restore the previous state
     let &filetype = old_filetype
@@ -982,6 +982,7 @@ endfunction
 " Get the filetype for the specified buffer
 function! s:Tlist_Get_Buffer_Filetype(bnum)
     let buf_ft = getbufvar(a:bnum, '&filetype')
+    let buf_ft = matchstr(buf_ft, '[^.]\+')
 
     if bufloaded(a:bnum)
         " For loaded buffers, the 'filetype' is already determined
